server/src/main/java/com/linecorp/centraldogma/server/internal/api/RepositoryServiceV1.java

```diff
@@ -186,8 +191,41 @@ public CompletableFuture<RepositoryDto> patchRepository(@Param String repoName,
      */
     @Get("/projects/{projectName}/repos/{repoName}/revision/{revision}")
     @RequiresReadPermission
-    public Map<String, Integer> normalizeRevision(Repository repository, @Param String revision) {
+    public Map<String, Integer> normalizeRevision(ServiceRequestContext ctx,
+                                                  Repository repository, @Param String revision) {
         final Revision normalizedRevision = repository.normalizeNow(new Revision(revision));
+        final Revision head = repository.normalizeNow(Revision.HEAD);
+        increaseCounterIfOldRevisionUsed(ctx, repository.parent().name(), repository.name(),
+                                         normalizedRevision, head);
         return ImmutableMap.of("revision", normalizedRevision.major());
     }
+
+    static void increaseCounterIfOldRevisionUsed(ServiceRequestContext ctx, Repository repository,
+                                                 Revision revision) {
+        final Revision normalized = repository.normalizeNow(revision);
+        final Revision head = repository.normalizeNow(Revision.HEAD);
+        increaseCounterIfOldRevisionUsed(ctx, repository.parent().name(), repository.name(), normalized, head);
+    }
+
+    public static void increaseCounterIfOldRevisionUsed(
+            ServiceRequestContext ctx, String projectName, String repoName,
+            Revision normalized, Revision head) {
+        if (normalized.major() == 1 || head.major() - normalized.major() >= 5000) {
+            ctx.log().whenComplete().thenAccept(log -> {
+                ctx.meterRegistry().counter("dogma.old.revision",
+                                            generateTags(projectName, repoName, log, normalized, head))
+                   .increment();
+            });
+        }
+    }
+
+    private static List<Tag> generateTags(String projectName, String repoName,
+                                          RequestLog log, Revision normalized, Revision head) {
+        return ImmutableList.of(Tag.of("project", projectName),
+                                Tag.of("repo", repoName),
+                                Tag.of("service", firstNonNull(log.serviceName(), "none")),
+                                Tag.of("method", log.name()),
+                                Tag.of("normalized", Integer.toString(normalized.major())),
```

Comment:
I have added the metric with the assumption that this will be short-lived and there will not be many requests that are caught by the condition.
However, there's no reason not to decrease the cardinality so let me use the distribution. Thanks!

Reply:
> I have added the metric with the assumption that this will be short-lived and there will not be many requests that are caught by the condition.

I agree, but I also thought this code will also need to go to production env. to confirm our hypothesis 😄